lib/secret_keys/cli.rb

```diff
@@ -0,0 +1,140 @@
+        end
+      end
+
+      def run
+      end
+    end
+
+    class Decrypt < CLI
+      def run
+      end
+    end
+
+    class Read < CLI
+      def parse_additional_options(opts)
+        @key = nil
+        opts.on("-k", "--key KEY", String, "Key from the file to output. You can use dot notation to read a nested key.") do |value|
+          @key = value
+        end
+      end
+
+      def run
+        raise ArgumentError.new("key is required") if @key.nil? || @key.empty?
+        val = secrets.to_h
+        @key.split(".").each do |key|
+          val = secrets[key] if val.is_a?(Hash)
+        end
+        val
+      end
+    end
+
+    class Edit < CLI
+      def parse_additional_options(opts)
+        @actions = []
+        opts.on("-e", "--set-encrypted KEY=VALUE", String, "Set an encrypted value in the file. You can use dot notation to set a nested value.") do |value|
+          key, val = value.split("=", 2)
+          @actions << [:encrypt, key, val]
+        end
+        opts.on("-d", "--set-decrypted KEY=VALUE", String, "Set a plain text value in the file. You can use dot notation to set a nested value.") do |value|
+          key, val = value.split("=", 2)
+          @actions << [:decrypt, key, val]
+        end
+        opts.on("-r", "--set-encrypted KEY=VALUE", String, "Remove a key from the file. You can use dot notation to remove a nested value.") do |value|
+          @actions << [:remove, value]
+        end
+      end
+
+      def run
+      end
+    end
+
+    attr_reader :secrets
+
+    def initialize(argv)
+      parse_options(argv)
+      @secrets = SecretKeys.new(@input, @secret_key)
+    end
+
+    # Subclasses can override this method to parse additional options beyond the standard set.
+    def parse_additional_options(opts)
+    end
+
+    # Subclasses must implement this method to execute the logic.
+    def run
+      raise NotImplementedError
+    end
+
+    private
+
+    def parse_options(argv)
+      @secret_key = nil
+      @format = nil
+
+      OptionParser.new do |opts|
+        opts.banner = "Usage: secret_keys encrypt|decrypt|read|edit [options] input_file|- output_file|-"
+
+        opts.on("--help", "Prints this help") do
+          puts opts.help
+          exit
+        end
+
+        opts.on("-s", "--secret-key ENCRYPTION_KEY", String, "Encryption key used to encrypt strings in the file. This value can also be passed in the SECRET_KEYS_ENCRYPTION_KEY environment variable or via STDIN by specifying -.") do |value|
+          @secret_key = get_secret_key(value)
+        end
+
+        opts.on("--secret-key-file ENCRYPTION_KEY_FILE_PATH", String, "Path to a file that contains the encryption key. This value can also be passed in the SECRET_KEYS_ENCRYPTION_KEY environment variable.") do |value|
+          @secret_key = File.read(value).chomp
+        end
+
+        opts.on("-f", "--format JSON|YAML", String, "Set the output format. By default this will be the same as the input format.") do |value|
```

Comment:
```suggestion
        opts.on("-f", "--format [FORMAT]", [:json, :yaml, :auto], "Set the output format. By default this will be the same as the input format.") do |value|
```
You can pass an array of valid values to choose from and OptionParser will do nice things for you.